### PR TITLE
samples: net: add 9151 support

### PR DESCRIPTION
**Pull request by DematteisGiacomo**
Add 9151 support to net samples.

### CODE REVIEW

**Review by maxd-nordic — APPROVED** [review]

File: samples/net/http_server/boards/nrf9151dk_nrf9151_ns.conf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Configuration file for nRF9161 DK
```
Comment (eivindj-nordic) [inline samples/net/http_server/boards/nrf9151dk_nrf9151_ns.conf:7]:
```suggestion
# Configuration file for nRF9151 DK
```

File: samples/net/mqtt/boards/nrf9151dk_nrf9151_ns.conf [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Configuration file for nRF9161 DK
```
Comment (eivindj-nordic) [inline samples/net/mqtt/boards/nrf9151dk_nrf9151_ns.conf:7]:
```suggestion
# Configuration file for nRF9151 DK
```

**Review by eivindj-nordic — APPROVED** [review]
Some nits, otherwise looks good!

**Comment by lemrey** [comment]
We should have some automation in place to make sure we at least build these samples for this new board.

**Comment by DematteisGiacomo** [comment]
> We should have some automation in place to make sure we at least build these samples for this new board. @DematteisGiacomo is there any kind of automation setup for these sample at the moment?

@lemrey yes there is, I am adding tests for it in our nightly. That is the reason why I am adding support for the board in the first place :+1:

**Comment by divipillai** [comment]
In the current release, do we also plan to add the nRF9151 board configuration in the sample.yml file for these samples?

**Comment by lemrey** [comment]
> In the current release, do we also plan to add the nRF9151 board configuration in the sample.yml file for these samples?

That's what I am after too! Great that we have nightly tests, but do we ensure that these samples keep building for this new board on-commit as well?

**Comment by DematteisGiacomo** [comment]
@lemrey @divipillai I made a tentative pr to address your comments: https://github.com/nrfconnect/sdk-nrf/pull/13810

**Pull request merged**